app/controllers/items_controller.rb

```diff
@@ -585,38 +659,59 @@ def apply_apo_defaults
   # add a workflow to an object if the workflow is not present in the active table
   def add_workflow
     return unless params[:wf]
-    wf = @object.workflows[params[:wf]]
-    # check for this workflow is present and active (not archived)
+    wf_name = params[:wf]
+    wf = @object.workflows[wf_name]
+    # check the workflow is present and active (not archived)
     if wf && wf.active?
-      render :status => 500, :text => "#{params[:wf]} already exists!"
+      render :status => 500, :text => "#{wf_name} already exists!"
       return
     end
-    @object.create_workflow(params[:wf])
+    @object.create_workflow(wf_name)
 
     # We need to sync up the workflows datastream with workflow service (using #find)
     # and then force a committed Solr update before redirection.
-    reindex Dor::Item.find(params[:id])
+    reindex find_druid(params[:id])
     Dor::SearchService.solr.commit
 
+    msg = "Added #{wf_name}"
     if params[:bulk]
-      render :text => "Added #{params[:wf]}"
+      render :text => msg
     else
-      redirect_to catalog_path(params[:id]), :notice => "Added #{params[:wf]}"
+      redirect_to catalog_path(params[:id]), notice: msg
     end
   end
 
   private
 
-  def reindex(item)
-    Dor::SearchService.solr.add item.to_solr
+  # check that the user can carry out this item modification
+  def allow_modify
+    return true if current_user.can_manage_object?(@object)
+    render :status => :forbidden, :text => 'forbidden'
+    false
+  end
+
+  def allow_view
+    return true if current_user.can_view_object?(@object)
+    render :status => :forbidden, :text => 'forbidden'
+    false
+  end
+
```

Comment:
The `allow_modify` and `allow_view` methods now make use of the methods consolidated in the User model.